File: Tutor Demos/Week 7/Java/Order.java
public class Order implements Request{

    private String name;
    private String products;

    public Order(String name, String products) {
        this.name = name;
        this.products = products;
    }

    public String getRequestLine() {
        return this.name + " has requested " + products;
    }

    public void handle() {
        for (String item : products.split(",")) {
            System.out.print("Item: " + item.trim() + ". ");
            char choice = readChoice();
            System.out.println(choice == 'y' ? "Approved" : "Rejected");
        }
    }

    private char readChoice() {
        System.out.print("Approve request? (y/n): ");
        return In.nextChar();
    }
}

File: Tutor Demos/Week 7/Python/order.py
from request import Request
class Order(Request):
    def __init__(self, name, products):
        self.__name = name
        self.__products = products
    
    def get_request_line(self):
        return f"{self.__name} has requested {self.__products}"
    
    def handle(self):
        for item in self.__products.split(","):
            print(f"Item: {item}. ")
            choice = input("Approve request? (y/n): ")
            print("Approved" if choice == "y" else "Rejected")
            